### PR TITLE
noneSatisfy should report all failing elements

### LINKED ISSUE #1415: noneSatisfy should report all failing elements
#### Summary
Currently, `noneSatisfy` does only report the first element which did not satisfy the given restrictions. 
What about reporting all elements which did not satisfy the given restrictions.

#### Example

A very simple example, e.g. let's assume that we have a list `[1, 2, 3]`. If we use noneSatisfy and let's say the 2nd and the 3rd element does not meet the given restrictions we only report the 2nd one.

```
java.lang.AssertionError: 
Expecting no elements of:
  <[1, 2, 3]>
to satisfy the given assertions requirements but this one did:
  <2>
```

### PR DESCRIPTION
**Pull request by epeee**
#### Check List:
* Fixes #1415 
* Unit tests : YES
* Javadoc with a code example (API only) : NA

The example from #1415 now results in:

```
java.lang.AssertionError: 
Expecting no elements of:
  <[1, 2, 3]>
to satisfy the given assertions requirements but these elements did:
  <[2, 3]>
```

### CODE REVIEW

File: src/test/java/org/assertj/core/internal/iterables/Iterables_assertNoneSatisfy_Test.java [hunk 1]
```diff
@@ -67,7 +67,18 @@ public void should_fail_when_one_elements_satisfy_the_given_restrictions() {
...
+    verify(failures).failure(info, noElementsShouldSatisfy(actual, list("Yoda")));
+    assertThat(assertionError).isNotNull();
+  }
+
+  @Test
+  public void should_fail_when_two_elements_satisfy_the_given_restrictions() {
+    // GIVEN
+    Consumer<String> restrictions = name -> assertThat(name).startsWith("L");
+    // WHEN
+    Throwable assertionError = catchThrowable(() -> iterables.assertNoneSatisfy(someInfo(), actual, restrictions));
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/iterables/Iterables_assertNoneSatisfy_Test.java]: can be replaced by `AssertionsUtil.expectAssertionError` that checks `assertThat(assertionError).isNotNull();` internally.

**Comment by joel-costigliola** [comment]
Looks good @epeee, I just had a few minor comments

File: src/test/java/org/assertj/core/internal/iterables/Iterables_assertNoneSatisfy_Test.java (hunk 1 above)
Comment (epeee) [reply src/test/java/org/assertj/core/internal/iterables/Iterables_assertNoneSatisfy_Test.java]: :+1:

File: src/test/java/org/assertj/core/error/NoElementsShouldSatisfy_create_Test.java [hunk 2]
```diff
@@ -16,24 +16,24 @@
...
 import org.junit.jupiter.api.Test;
 
 public class NoElementsShouldSatisfy_create_Test {
 
   @Test
-  public void should_create_error_message_any() {
+  public void should_create_error_message() {
     // GIVEN
-    ErrorMessageFactory factory = noElementsShouldSatisfy(newArrayList("Luke", "Yoda"), "Vader");
+    ErrorMessageFactory factory = noElementsShouldSatisfy(list("Luke", "Leia", "Yoda"), list("Luke", "Leia"));
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/error/NoElementsShouldSatisfy_create_Test.java]: not sure why but I had comment here to put a `%` to check that reserved `String.format` flags did not break the error message construction

**Comment by joel-costigliola** [comment]
thanks one more comment and we good!

File: src/test/java/org/assertj/core/error/NoElementsShouldSatisfy_create_Test.java (hunk 2 above)
Comment (epeee) [reply src/test/java/org/assertj/core/error/NoElementsShouldSatisfy_create_Test.java]: I've added a testcase using a `%` and it looks like we are fine.

**Comment by joel-costigliola** [comment]
Integrated thanks @epeee!

FYI I slightly improved the test I was commenting on this way:

```
  @Test
  public void should_fail_when_two_elements_satisfy_the_given_restrictions() {
    // GIVEN
    Consumer<String> restrictions = name -> assertThat(name).startsWith("L");
    // WHEN
    expectAssertionError(() -> iterables.assertNoneSatisfy(someInfo(), actual, restrictions));
    // THEN
    verify(failures).failure(info, noElementsShouldSatisfy(actual, list("Luke", "Leia")));
  }
```
No need to check the exception since this is done inside `expectAssertionError`.

**Pull request closed without merging**